sql-plugin/src/main/scala/org/apache/spark/sql/rapids/predicates.scala

```diff
@@ -316,6 +316,79 @@ case class GpuEqualNullSafe(left: Expression, right: Expression) extends CudfBin
   }
 }
 
+case class GpuEqualToNoNans(left: Expression, right: Expression) extends CudfBinaryComparison
+    with NullIntolerant {
+  override def symbol: String = "="
+  override def outputTypeOverride: DType = DType.BOOL8
+  override def binaryOp: BinaryOp = BinaryOp.EQUAL
+
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuColumnVector): ColumnVector = {
+    val result = super.doColumnar(lhs, rhs)
+    if (hasFloatingPointInputs) {
+      withResource(result) { result =>
+        withResource(lhs.getBase.isNan) { lhsNan =>
+          withResource(rhs.getBase.isNan) { rhsNan =>
+            withResource(lhsNan.or(rhsNan)) { lhsNanOrRhsNan =>
+              withResource(lhsNanOrRhsNan.not) { neitherNan =>
+                neitherNan.and(result)
+              }
+            }
+          }
+        }
+      }
+    } else {
+      result
+    }
+  }
+
+  override def doColumnar(lhs: GpuScalar, rhs: GpuColumnVector): ColumnVector = {
+    val result = super.doColumnar(lhs, rhs)
+    if (hasFloatingPointInputs) {
+      withResource(result) { result =>
+        withResource(Scalar.fromBool(lhs.isNan)) { lhsNan =>
+          withResource(rhs.getBase.isNan) { rhsNan =>
+            withResource(lhsNan.or(rhsNan)) { lhsNanOrRhsNan =>
+              withResource(lhsNanOrRhsNan.not) { neitherNan =>
+                neitherNan.and(result)
+              }
+            }
+          }
+        }
+      }
+    } else {
+      result
+    }
+  }
+
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuScalar): ColumnVector = {
+    val result = super.doColumnar(lhs, rhs)
+    if (hasFloatingPointInputs) {
+      withResource(result) { result =>
+        withResource(lhs.getBase.isNan) { lhsNan =>
+          withResource(Scalar.fromBool(rhs.isNan)) { rhsNan =>
+            withResource(lhsNan.or(rhsNan)) { lhsNanOrRhsNan =>
+              withResource(lhsNanOrRhsNan.not) { neitherNan =>
+                neitherNan.and(result)
+              }
+            }
+          }
+        }
+      }
+    } else {
+      result
+    }
+  }
+
+  override def convertToAst(numFirstTableColumns: Int): ast.AstExpression = {
```

Comment:
I don't think we have a way to reach this code currently. We don't support AST in aggregation projections like that, so I think it would be cleaner to just delete this method and let the default implementation throw an exception if it is called.